xwiki-platform-core/xwiki-platform-model/xwiki-platform-model-api/src/main/java/org/xwiki/model/internal/reference/AbstractStringEntityReferenceResolver.java

```diff
@@ -115,6 +115,18 @@ private StringBuilder createStringBuilder(String entityReferenceRepresentation)
     @Override
     public EntityReference resolve(String entityReferenceRepresentation, EntityType type, Object... parameters)
     {
+        // If the type is not provided, try to extract it from the string
+        if (type == null) {
+            Character entityTypeSeparator = getSymbolScheme().getEntityTypeSeparator();
+            if (entityTypeSeparator != null) {
+                type = Enum.valueOf(EntityType.class,
```

Comment:
Feels strange to assign a new value to a parameter.

Reply:
It's simply that we try to find it if it's null. I guess we could define new variable which by default have the parameters values, but I'm not sure it would really help readability in any way (might make it worst).

Reply:
No complain from sonar or checkstyle for that? I thought there were checks for those.

Reply:
Yes, I was surprised too, but neither checkstyle (during the Maven build) nor sonarlint (in Eclipse) are complaining about it for me.